llvm/lib/IR/RepoTicket.cpp

```diff
@@ -25,6 +26,14 @@
 
 using namespace llvm;
 
+#define DEBUG_TYPE "prepo-ticket"
+
+STATISTIC(NumFunctions, "Number of functions hashed");
```

Comment:
The `STATISTIC` macro seems to originate in Statistic.h with a family of support functions (e.g. `PrintStatistics()`). However, we're just using `LLVM_DEBUG()` to dump them (line #267). Is this right?

Reply:
Yes, it is right. From my understanding, the `PrintStatistics()` function prints all the LLVM statistics. However, we only want to dump the statistic variable defined in this Pass when the `debug-only` is enabled. Therefore, we are using “LLVM_DEBUG”.

Reply:
> we only want to dump the statistic variable defined in this Pass when the debug-only is enabled. Therefore, we are using “LLVM_DEBUG”.

If that’s what we want, then why switch to using the `STATISTIC` macro?

Reply:
> If that’s what we want, then why switch to using the `STATISTIC` macro?

I think that a STATISTIC is probably useful for other types of reporting as well (opt -stats I think). Could we use opt -stats in the tests as well? I think that will only happen when assertions are enabled so (I think) should do what you want.

Reply:
Yes, We could. I have removed the `LLVM_DEBUG` and updated the repo tests correspondingly. I will commit later with other changes.